src/vmm/src/builder.rs

```diff
@@ -453,7 +472,11 @@ fn load_kernel(
         kernel::loader::load_kernel(guest_memory, &mut kernel_file, arch::get_kernel_start())
             .map_err(StartMicrovmError::KernelLoader)?;
 
-    Ok(entry_addr)
+    // The program headers of the kernel image are necessary in the address translation
+    // mechanism in the GDB Server thread
+    let phdrs = kernel::loader::extract_phdrs(&mut kernel_file).unwrap();
```

Comment:
To me it seems this information could be also extracted later at runtime, in the RPC interface no?